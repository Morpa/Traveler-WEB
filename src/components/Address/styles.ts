import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Header = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: baseline;

    border-bottom: 1px solid ${theme.colors.secondary};
    padding: ${theme.spacings.xsmall} 0;

    > h1 {
      color: ${theme.colors.titleColor};
    }

    > a {
      margin-left: 16rem;
      text-decoration: none;
      color: ${theme.colors.textComplementary};
      font-family: 'Heebo', sans-serif;
    }
  `}
`
export const Map = styled.div`
  ${({ theme }) => css`
    height: 16rem;
    width: 45rem;
    margin-top: ${theme.spacings.medium};
  `}
`
export const Address = styled.p`
  ${({ theme }) => css`
    font-family: 'Heebo', sans-serif;
    font-weight: ${theme.font.light};
    margin-top: ${theme.spacings.xsmall};
    color: ${theme.colors.textColor};
  `}
`
