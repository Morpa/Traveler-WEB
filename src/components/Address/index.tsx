import Link from 'next/link'
import dynamic from 'next/dynamic'

import * as S from './styles'

export type AddressProps = {
  address: string
  coords: number[]
}

const MapWithNoSSR = dynamic(() => import('../Map/Map'), {
  ssr: false
})

const Address = ({ address, coords }: AddressProps) => (
  <S.Wrapper>
    <S.Header>
      <h1>EndereÃ§o</h1>
      <Link href="/" passHref>
        <a>Ver no Google Maps</a>
      </Link>
    </S.Header>
    <S.Map>
      <MapWithNoSSR coords={coords} />
    </S.Map>
    <S.Address>{address}</S.Address>
  </S.Wrapper>
)

export default Address
