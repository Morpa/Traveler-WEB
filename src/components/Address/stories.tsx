import { Story, Meta } from '@storybook/react/types-6-0'
import Address, { AddressProps } from '.'

export default {
  title: 'Address',
  component: Address,
  args: {
    address: 'Av. da Boavista, 850 Nevolgide, Porto - PT, 8801-290',
    coords: [41.1621376, -8.6569731]
  }
} as Meta

export const Default: Story<AddressProps> = (args) => (
  <div style={{ maxWidth: '45rem', maxHeight: '16rem' }}>
    <Address {...args} />
  </div>
)
