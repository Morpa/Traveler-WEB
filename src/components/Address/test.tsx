import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Address from '.'

const props = {
  address: 'Av. da Boavista, 850 Nevolgide, Porto - PT, 8801-290',
  coords: [41.1621376, -8.6569731]
}

describe('<Address />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Address {...props} />)

    expect(
      screen.getByRole('heading', { name: /Endere√ßo/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
