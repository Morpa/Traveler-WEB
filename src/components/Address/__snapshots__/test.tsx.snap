// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-bottom: 1px solid #DCE2E6;
  padding: 1.6rem 0;
}

.c0 > h1 {
  color: #123952;
}

.c0 > a {
  margin-left: 16rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #A0ACB3;
  font-family: 'Heebo',sans-serif;
}

.c1 {
  height: 16rem;
  width: 45rem;
  margin-top: 3.2rem;
}

.c2 {
  font-family: 'Heebo',sans-serif;
  font-weight: 300;
  margin-top: 1.6rem;
  color: #617480;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <h1>
      Endere√ßo
    </h1>
    <a
      href="/"
    >
      Ver no Google Maps
    </a>
  </div>
  <div
    class="c1"
  />
  <p
    class="c2"
  >
    Av. da Boavista, 850 Nevolgide, Porto - PT, 8801-290
  </p>
</div>
`
